include/SFML/System/Vector2.hpp

```diff
@@ -68,14 +73,132 @@ class Vector2
+    
+    ////////////////////////////////////////////////////////////
+    /// \brief Length of the vector <i><b>(floating-point)</b></i>.
+    ///
+    /// If you are not interested in the actual length, but only in comparisons, consider using lengthSq().
+    ///
+    ////////////////////////////////////////////////////////////
+    T length() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Square of vector's length <i><b>(floating-point)</b></i>.
+    /// 
+    /// Suitable for comparisons, more efficient than length().
+    ///
+    ////////////////////////////////////////////////////////////
+    T lengthSq() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Vector with same direction but length 1 <i><b>(floating-point)</b></i>.
+    /// 
+    /// \pre \c vector is no zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2 normalized() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Signed angle from \c *this to \c rhs <i><b>(floating-point)</b></i>.
+    /// 
+    /// \return Angle in degrees in the interval [-180,180].
+    /// The angle determines how much you have to rotate \c *this
+    ///  until it points to the same direction as \c rhs.
+    /// \pre Neither \c *this nor \c rhs is a zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Angle signedAngleTo(const Vector2& rhs) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Angle from +X or (1,0) vector <i><b>(floating-point)</b></i>.
+    /// 
+    /// The vector (1,0) corresponds to 0 degrees, (0,1) corresponds to 90 degrees.
+    /// 
+    /// \return Angle in degrees in the interval [-180,180].
+    /// \pre \c vector is no zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Angle polarAngle() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Rotate by given angle <i><b>(floating-point)</b></i>.
+    /// 
+    /// The vector (1,0) corresponds 0 degrees, (0,1) corresponds 90 degrees.
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2 rotatedBy(Angle angle) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Projection of \c vector onto \c axis <i><b>(floating-point)</b></i>.
+    /// 
+    /// \param vector Vector to project.
+    /// \param axis Vector being projected onto. Need not be normalized, but must not have length zero.
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2 projectedOnto(const Vector2& axis) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Returns a perpendicular vector.
+    /// 
+    /// Returns \c vector rotated by 90 degrees counter clockwise; (x,y) becomes (-y,x).
```

Comment:
> counter clockwise

It does not really make sense to me. That depends on the coordinate system.
Maybe using the word "positive" would be better?

Reply:
You're right, I'll change it.

The majority of SFML code uses the convention where +X is right and +Y is down; should we mention this somewhere? Or remain completely agnostic to the coordinate system, but also make things a bit harder to the user?

Maybe `sf::Vector2` doc is not the right place for the coordinate system conventions; rather the tutorials or so.

Reply:
I added a line "in SFML's default coordinate system, ..." to make things explicit.

Reply:
Good, but it is clockwise then. :wink:

Reply:
You got me there 😬 thanks!